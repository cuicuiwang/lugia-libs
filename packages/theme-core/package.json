{
  "name": "@lugia/theme-core",
  "version": "1.0.36",
  "description": "Theme core",
  "main": "./lib/index.js",
  "scripts": {
    "start": "node ./lib"
  },
  "author": "#lugia#",
  "license": "MIT",
  "dependencies": {
    "@babel/runtime": "^7.4.5",
    "@lugia/object-utils": "^1.0.15"
  },
  "devDependencies": {
    "@lugia/theme-hoc-devtools": "^1.0.8"
  },
  "peerDependencies": {
    "@lugia/theme-hoc-devtools": "1.0.5"
  },
  "files": [
    "lib",
    "interface",
    "yarn.lock"
  ],
  "engines": {
    "node": ">=10",
    "npm": ">=5.2.0"
  }
}
