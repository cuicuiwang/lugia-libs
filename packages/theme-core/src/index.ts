/**
 *
 * create by ligx
 *
 * @flow
 */
import {
  addFocusBlurEvent,
  addMouseEvent,
  CSSComponentContainerDisplayName,
  CSSComponentDisplayName,
  filterSelector,
  getConfig,
  getKeys,
  getMatchSelector,
  getObject,
  hasThemeStateEvent,
  injectThemeStateEvent,
  selectThemeMeta,
  selectThemePart,
  ThemeComponentPrefix,
} from './utils';
import ThemeHandle, { packDisplayName } from './ThemeHandle';
import ThemeStateHandle from './ThemeStateHandle';
import * as React from 'react';

export const ThemeContext: any = React.createContext({});
export {
  packDisplayName,
  ThemeHandle,
  ThemeStateHandle,
  filterSelector,
  getKeys,
  getObject,
  getConfig,
  getMatchSelector,
  selectThemePart,
  selectThemeMeta,
  CSSComponentDisplayName,
  CSSComponentContainerDisplayName,
  ThemeComponentPrefix,
  addFocusBlurEvent,
  addMouseEvent,
  injectThemeStateEvent,
  hasThemeStateEvent,
};
