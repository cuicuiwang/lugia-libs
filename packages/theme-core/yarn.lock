# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


"@babel/runtime@^7.4.5":
  version "7.4.5"
  resolved "https://registry.npm.taobao.org/@babel/runtime/download/@babel/runtime-7.4.5.tgz?cache=0&other_urls=https%3A%2F%2Fregistry.npm.taobao.org%2F%40babel%2Fruntime%2Fdownload%2F%40babel%2Fruntime-7.4.5.tgz#582bb531f5f9dc67d2fcb682979894f75e253f12"
  integrity sha1-WCu1MfX53GfS/LaCl5iU914lPxI=
  dependencies:
    regenerator-runtime "^0.13.2"

"@lugia/object-utils@^1.0.15":
  version "1.0.15"
  resolved "https://registry.yarnpkg.com/@lugia/object-utils/-/object-utils-1.0.15.tgz#ac1dbf6fdaee696d2adb7069e75621dbaf808084"
  integrity sha512-03FD7ACx2aXS1NZkpXfQODpHLimLxqHQoCtBvSGhyV6I9VEUBklMnsG0NeD9GTrXg4rzywVGK/6xeWSoA7nIfw==
  dependencies:
    "@babel/runtime" "^7.4.5"
    deepmerge "^3.2.0"
    is-plain-object "^3.0.0"

"@lugia/react-devtools-core@^1.0.3":
  version "1.0.3"
  resolved "https://registry.yarnpkg.com/@lugia/react-devtools-core/-/react-devtools-core-1.0.3.tgz#725ba0178f3ebf946beaec19675de6c4049a3220"
  integrity sha512-yrzu/WRj5Vf/5ZxyBv6WPjdRG/UWlvqCpVzNkniAHGm5y+nJbRKUOeJDAyJoDMdogy3m1GLrxIbafjTvhQZg2Q==
  dependencies:
    shell-quote "^1.6.1"
    ws "^3.3.1"

"@lugia/theme-hoc-devtools@^1.0.8":
  version "1.0.8"
  resolved "https://registry.yarnpkg.com/@lugia/theme-hoc-devtools/-/theme-hoc-devtools-1.0.8.tgz#b957144cce86939b8ab04028a52bef22d54ddf1e"
  integrity sha512-DNF2g+V7NQuToV5OLiqbZPe8GGgFZqe8aKQrGzB+X/6PD/8Hno/KsDbfZM1shGh6w5HgJbYZ1mwpJng8+SCzWA==
  dependencies:
    "@lugia/react-devtools-core" "^1.0.3"

async-limiter@~1.0.0:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/async-limiter/-/async-limiter-1.0.1.tgz#dd379e94f0db8310b08291f9d64c3209766617fd"
  integrity sha512-csOlWGAcRFJaI6m+F2WKdnMKr4HhdhFVBk0H/QbJFMCr+uO2kwohwXQPxw/9OCxp05r5ghVBFSyioixx3gfkNQ==

debug@^4.1.1:
  version "4.1.1"
  resolved "https://registry.yarnpkg.com/debug/-/debug-4.1.1.tgz#3b72260255109c6b589cee050f1d516139664791"
  integrity sha512-pYAIzeRo8J6KPEaJ0VWOh5Pzkbw/RetuzehGM7QRRX5he4fPHx2rdKMB256ehJCkX+XRQm16eZLqLNS8RSZXZw==
  dependencies:
    ms "^2.1.1"

deepmerge@^3.2.0:
  version "3.3.0"
  resolved "https://registry.yarnpkg.com/deepmerge/-/deepmerge-3.3.0.tgz#d3c47fd6f3a93d517b14426b0628a17b0125f5f7"
  integrity sha512-GRQOafGHwMHpjPx9iCvTgpu9NojZ49q794EEL94JVEw6VaeA8XTUyBKvAkOOjBX9oJNiV6G3P+T+tihFjo2TqA==

is-plain-object@^3.0.0:
  version "3.0.0"
  resolved "https://registry.yarnpkg.com/is-plain-object/-/is-plain-object-3.0.0.tgz#47bfc5da1b5d50d64110806c199359482e75a928"
  integrity sha512-tZIpofR+P05k8Aocp7UI/2UTa9lTJSebCXpFFoR9aibpokDj/uXBsJ8luUu0tTVYKkMU6URDUuOfJZ7koewXvg==
  dependencies:
    isobject "^4.0.0"

isobject@^4.0.0:
  version "4.0.0"
  resolved "https://registry.yarnpkg.com/isobject/-/isobject-4.0.0.tgz#3f1c9155e73b192022a80819bacd0343711697b0"
  integrity sha512-S/2fF5wH8SJA/kmwr6HYhK/RI/OkhD84k8ntalo0iJjZikgq1XFvR5M8NPT1x5F7fBwCG3qHfnzeP/Vh/ZxCUA==

ms@^2.1.1:
  version "2.1.2"
  resolved "https://registry.yarnpkg.com/ms/-/ms-2.1.2.tgz#d09d1f357b443f493382a8eb3ccd183872ae6009"
  integrity sha512-sGkPx+VjMtmA6MX27oA4FBFELFCZZ4S4XqeGOXCv68tT+jb3vk/RyaKWP0PTKyWtmLSM0b+adUTEvbs1PEaH2w==

regenerator-runtime@^0.13.2:
  version "0.13.2"
  resolved "https://registry.npm.taobao.org/regenerator-runtime/download/regenerator-runtime-0.13.2.tgz?cache=0&other_urls=https%3A%2F%2Fregistry.npm.taobao.org%2Fregenerator-runtime%2Fdownload%2Fregenerator-runtime-0.13.2.tgz#32e59c9a6fb9b1a4aff09b4930ca2d4477343447"
  integrity sha1-MuWcmm+5saSv8JtJMMotRHc0NEc=

safe-buffer@~5.1.0:
  version "5.1.2"
  resolved "https://registry.yarnpkg.com/safe-buffer/-/safe-buffer-5.1.2.tgz#991ec69d296e0313747d59bdfd2b745c35f8828d"
  integrity sha512-Gd2UZBJDkXlY7GbJxfsE8/nvKkUEU1G38c1siN6QP6a9PT9MmHB8GnpscSmMJSoF8LOIrt8ud/wPtojys4G6+g==

shell-quote@^1.6.1:
  version "1.7.2"
  resolved "https://registry.yarnpkg.com/shell-quote/-/shell-quote-1.7.2.tgz#67a7d02c76c9da24f99d20808fcaded0e0e04be2"
  integrity sha512-mRz/m/JVscCrkMyPqHc/bczi3OQHkLTqXHEFu0zDhK/qfv3UcOA4SVmRCLmos4bhjr9ekVQubj/R7waKapmiQg==

ultron@~1.1.0:
  version "1.1.1"
  resolved "https://registry.yarnpkg.com/ultron/-/ultron-1.1.1.tgz#9fe1536a10a664a65266a1e3ccf85fd36302bc9c"
  integrity sha512-UIEXBNeYmKptWH6z8ZnqTeS8fV74zG0/eRU9VGkpzz+LIJNs8W/zM/L+7ctCkRrgbNnnR0xxw4bKOr0cW0N0Og==

ws@^3.3.1:
  version "3.3.3"
  resolved "https://registry.yarnpkg.com/ws/-/ws-3.3.3.tgz#f1cf84fe2d5e901ebce94efaece785f187a228f2"
  integrity sha512-nnWLa/NwZSt4KQJu51MYlCcSQ5g7INpOrOMt4XV8j4dqTXdmlUmSHQ8/oLC069ckre0fRsgfvsKwbTdtKLCDkA==
  dependencies:
    async-limiter "~1.0.0"
    safe-buffer "~5.1.0"
    ultron "~1.1.0"
