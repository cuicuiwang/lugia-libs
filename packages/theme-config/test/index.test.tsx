// @flow

import * as React from 'react';
import Theme from '../src';
import renderer from 'react-test-renderer';
import { ThemeContext } from '@lugia/theme-core';
import { useContext } from 'react';

describe('ThemeConfig', () => {
  it('Theme', () => {
    const Test = (props: any) => {
      const context = useContext(ThemeContext);
      return (
        <div>
          {JSON.stringify(context)}
          {props.children}
        </div>
      );
    };
    const config = {
      hello: {
        test: {},
      },
    };
    const cmp = renderer.create(
      <Theme config={config}>
        <Test>hello everyone</Test>
      </Theme>,
    );
    expect(cmp).toMatchSnapshot();
  });
});
