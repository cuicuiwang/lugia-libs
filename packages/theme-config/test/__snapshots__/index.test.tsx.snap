// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeConfig Theme 1`] = `
<div>
  {"config":{"hello":{"test":{}}},"svThemeConfigTree":{"hello":{"test":{}}}}
  hello everyone
</div>
`;
